operator/pkg/apis/validation/validation.go

```diff
@@ -125,6 +125,14 @@ func detectCniIncompatibility(client kube.Client, cniEnabled bool, ztunnelEnable
 			errs = util.AppendErr(errs,
 				errors.New("detected Cilium CNI with 'bpf-lb-sock=true'; this requires 'bpf-lb-sock-hostns-only=true' to be set"))
 		}
+		kpr := cilium.Data["kube-proxy-replacement"]
+		if kpr == "strict" {
+			if !bpfLbHostnsOnly {
+				errs = util.AppendErr(errs,
+					errors.New("detected Cilium CNI with 'kube-proxy-replacement=strict' and 'bpf-lb-sock-hostns-only=false'; "+
```

Comment:
So is this basically saying that Cilium implicitly turns on `bpf-lb-sock` when you set `kube-proxy-replacement=strict`?

I don't know I see this...

```
$ cilium install --set kubeProxyReplacement=true  --set k8sServiceHost=cil-control-plane --set k8sServicePort=6443 --dry-run | rg bpf-lb
  # bpf-lb-map-max specifies the maximum number of entries in bpf lb service,
  bpf-lb-map-max: "65536"
  bpf-lb-external-clusterip: "false"
  bpf-lb-sock: "false"
```

Version v0.16.24